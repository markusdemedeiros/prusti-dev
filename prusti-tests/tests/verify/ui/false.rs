// compile-flags: -Pprint_desugared_specs=true -Pprint_typeckd_specs=true -Phide_uuids=true
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"
// normalize-stdout-test: "\[[a-z0-9]{4}\]::" -> "[$(CRATE_ID)]::"
// normalize-stdout-test: "#\[prusti::specs_version = \x22.+\x22\]" -> "#[prusti::specs_version = $(SPECS_VERSION)]"

use prusti_contracts::*;

#[ensures(false)]
fn test1() {}

fn test2() {
    assert!(false);
}

fn main() {}
