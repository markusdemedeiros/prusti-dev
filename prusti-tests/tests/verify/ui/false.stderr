error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/false.rs:9:11
   |
9  | #[ensures(false)]
   |           ^^^^^
   |
note: the error originates here
  --> $DIR/false.rs:10:1
   |
10 | fn test1() {}
   | ^^^^^^^^^^^^^

error: [Prusti: verification error] the asserted expression might not hold
  --> $DIR/false.rs:13:5
   |
13 |     assert!(false);
   |     ^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 2 previous errors

